django_lightweight_queue/app_settings.py

```diff
@@ -1,52 +1,58 @@
-    'django_lightweight_queue.middleware.transaction.TransactionMiddleware',
-))  # type: Sequence[str]
-
-# Apps to ignore when looking for tasks. Apps must be specified as the dotted
-# name used in `INSTALLED_APPS`. This is expected to be useful when you need to
-# have a file called `tasks.py` within an app, but don't want
-# django-lightweight-queue to import that file.
-# Note: this _doesn't_ prevent tasks being registered from these apps.
-IGNORE_APPS = setting('IGNORE_APPS', ())  # type: Sequence[str]
-
-# Backend-specific settings
-REDIS_HOST = setting('REDIS_HOST', '127.0.0.1')  # type: str
-REDIS_PORT = setting('REDIS_PORT', 6379)  # type: int
-REDIS_PASSWORD = setting('REDIS_PASSWORD', None)  # type: Optional[str]
-REDIS_PREFIX = setting('REDIS_PREFIX', '')  # type: str
-
-ENABLE_PROMETHEUS = setting('ENABLE_PROMETHEUS', False)  # type: bool
-# Workers will export metrics on this port, and ports following it
-PROMETHEUS_START_PORT = setting('PROMETHEUS_START_PORT', 9300)  # type: int
-
-ATOMIC_JOBS = setting('ATOMIC_JOBS', True)  # type: bool
+class Settings:
+    _uses_short_names: bool = True  # used later in checking for values
+
+
+class Defaults(Settings):
+    WORKERS: Dict[QueueName, int] = {}
+    BACKEND: str = 'django_lightweight_queue.backends.synchronous.SynchronousBackend'
+    LOGGER_FACTORY: Union[str, Callable[[str], Logger]] = 'logging.getLogger'
+    # Allow per-queue overrides of the backend.
+    BACKEND_OVERRIDES: Dict[QueueName, str] = {}
+    MIDDLEWARE: Sequence[str] = ('django_lightweight_queue.middleware.logging.LoggingMiddleware',)
+    # Apps to ignore when looking for tasks. Apps must be specified as the dotted
+    # name used in `INSTALLED_APPS`. This is expected to be useful when you need to
+    # have a file called `tasks.py` within an app, but don't want
+    # django-lightweight-queue to import that file.
+    # Note: this _doesn't_ prevent tasks being registered from these apps.
+    IGNORE_APPS: Sequence[str] = ()
+    REDIS_HOST: str = '127.0.0.1'
+    REDIS_PORT: int = 6379
+    REDIS_PASSWORD: Optional[str] = None
+    REDIS_PREFIX: str = ""
+    ENABLE_PROMETHEUS: bool = False
+    # Workers will export metrics on this port, and ports following it
+    PROMETHEUS_START_PORT: int = 9300
+    ATOMIC_JOBS: bool = True
+
+
+class AppSettings:
+    def __init__(self, layers: List[Settings]) -> None:
+        self._layers = layers
+
+    def add_layer(self, layer: Settings) -> None:  # to be called by `load_extra_config`
+        self._layers.append(layer)
+
+    def __getattr__(self, name: str) -> Any:
+        # reverse so that later layers override earlier ones
+        for layer in reversed(self._layers):
+            # check to see if the layer is internal or external
+            use_short_names = getattr(layer, "_uses_short_names", False)
+            attr_name = (
+                name if use_short_names else
+                '{}{}'.format(constants.SETTING_NAME_PREFIX, name)
+            )
```

Comment:
If it's all the same to you, I'd prefer to keep the adapting where it is; the more obfuscated magic there is, the harder it is to debug issues. This is already a fairly complex system, so while performing the modification automatically would be cleaner for this section, it could feasibly lead to frustration and confusion for new contributors trying to interact with this code later on.